tests/e2e/gc_mgmt_wrkld_domain_isolation.go

```diff
@@ -0,0 +1,351 @@
+	})
+
+	ginkgo.AfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		ginkgo.By(fmt.Sprintf("Deleting service nginx in namespace: %v", namespace))
+		err := client.CoreV1().Services(namespace).Delete(ctx, servicename, *metav1.NewDeleteOptions(0))
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		dumpSvcNsEventsOnTestFailure(client, namespace)
+
+		framework.Logf("Collecting supervisor PVC events before performing PV/PVC cleanup")
+		eventList, err := client.CoreV1().Events(namespace).List(ctx, metav1.ListOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, item := range eventList.Items {
+			framework.Logf(item.Message)
+		}
+	})
+
+	/*
+		TKG - Testcase-4
+		Dynamic and Pre-Provisioned snapshot creation with removal of zones from the namespace
+
+		Test Steps:
+		1. The expectation is that TKG worker nodes are spread across zones (zone-2, zone-3 and zone-4)
+		2. Create statefulset with replica count 3 and affinity set
+		such that each volume,
+		pod should comeup on each worker node.
+		3. Now, Mark zone-3 for removal
+		4. Increase the replica count from 3 to 6.
+		5. Verify if newly created pvcs,pod reach Bound or running state.
+		6. Now, take a volume snaphot of any 2 statefulset volumes.
+		7. Verify snapshot created successfully.
+		8. Create a static snapshot of any 1 snapshot created above.
+		9. Verify static snapshot on tkg created successfully.
+		10. Perform scaling operation. Increase replica count to 8
+		11. Verify scaling operation went smooth.
+		12. Perfrom cleanup: Delete Pods, volumes.
+	*/
+
+	ginkgo.It("Dynamic and Pre-Provisioned snapshot creation with removal of zones from the namespace", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// statefulset replica count
+		replicas = 3
+
+		ginkgo.By("Read shared storage policy tagged to wcp namespace")
+		storageclass, err := client.StorageV1().StorageClasses().Get(ctx, sharedStoragePolicyName, metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+
+		ginkgo.By("Creating statefulset")
+		statefulset := createCustomisedStatefulSets(ctx, client, namespace, true, replicas, true, allowedTopologies,
+			true, true, "", "", storageclass, storageclass.Name)
+		defer func() {
+			fss.DeleteAllStatefulSets(ctx, client, namespace)
+		}()
+
+		ginkgo.By("Verify svc pv affinity, pvc annotation and pod node affinity")
+		err = verifyPvcAnnotationPvAffinityPodAnnotationInSvc(ctx, client, statefulset, nil, nil, namespace,
+			allowedTopologies)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Mark zone-2 for removal from wcp namespace")
+		err = markZoneForRemovalFromWcpNs(vcRestSessionId, gcNamespace,
```

Comment:
Above, you’ve used wcpNamespace in the heading, but later you mention gcNamespace. Please correct this for consistency.